homeassistant/components/switch/thinkingcleaner.py

```diff
@@ -0,0 +1,96 @@
+"""Support for ThinkingCleaner."""
+import logging
+from datetime import timedelta
+
+import homeassistant.util as util
+from homeassistant.helpers.entity import ToggleEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['https://github.com/TheRealLink/pythinkingcleaner'
+                '/archive/v0.0.2.zip'
+                '#pythinkingcleaner==0.0.2']
+
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
+MIN_TIME_BETWEEN_FORCED_SCANS = timedelta(milliseconds=100)
+
+MIN_TIME_TO_WAIT = timedelta(seconds=5)
+
+SWITCH_TYPES = {
+    'clean': ['Clean', None, None],
+    'dock': ['Dock', None, None],
+    'find': ['Find', None, None],
+}
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the ThinkingCleaner platform."""
+    from pythinkingcleaner import Discovery
+
+    discovery = Discovery()
+    devices = discovery.discover()
+
+    @util.Throttle(MIN_TIME_BETWEEN_SCANS, MIN_TIME_BETWEEN_FORCED_SCANS)
+    def update_devices():
+        """Update all devices."""
+        for device_object in devices:
+            device_object.update()
+
+    dev = []
+    for device in devices:
+        for type_name in SWITCH_TYPES.keys():
+            dev.append(ThinkingCleanerSwitch(device, type_name,
+                                             update_devices))
+
+    add_devices(dev)
+
+
+class ThinkingCleanerSwitch(ToggleEntity):
+    """ThinkingCleaner Switch (dock, clean, find me)."""
+
+    def __init__(self, tc_object, switch_type, update_devices):
+        """Initialize the ThinkingCleaner."""
+        self.type = switch_type
+
+        self._update_devices = update_devices
+        self._tc_object = tc_object
+        self._state = \
+            self._tc_object.is_cleaning if switch_type == 'clean' else False
+        self.lock = False
+
+    @util.Throttle(MIN_TIME_TO_WAIT)
+    def toggle_lock(self):
+        """Lock the update since TC clean takes some time to update."""
+        self.lock = not self.lock
+        if self.lock:
+            self.toggle_lock()
```

Comment:
ah okay, i thought it would be pushed into a queue and will be executed after the delay.
the reason i added it was to keep the switch "on" since the state takes some time to update

is there a nice way to add that behavior?